### PR TITLE
User case 6 list updates user

### PR DESCRIPTION
**Pull request by phantranhieunhan**
User case 6 list updates user

### CODE REVIEW

File: module/friendship/adapter/postgres/repository/subscription.go [hunk 1]
```diff
@@ -69,3 +70,23 @@ func (s SubscriptionRepository) GetSubscription(ctx context.Context, ss domain.S
 	}
 	return convert.ToSubscriptionsDomain(m), nil
 }
+
+func (s SubscriptionRepository) GetSubscriptionEmailsByUserIDAndStatus(ctx context.Context, id string, status domain.SubscriptionStatus) ([]string, error) {
+	list := make([]view.SubscriberEmail, 0)
+	err := model.NewQuery(
+		qm.Select("u.email as subscriber_email"),
+		qm.From("subscriptions s"),
+		qm.LeftOuterJoin("users u on s.subscriber_id = u.id"),
```
Comment (nhan1603) [inline module/friendship/adapter/postgres/repository/subscription.go:79]: Why do we need to use `LeftOuterJoin` here ? 
As a foreign key is introduce, deletion from `user` table should be perform as `DELETE CASCADE` which will lead to the deletion of other related tables.

File: module/friendship/app/query/list_updates_user.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	// get list subscription from userId
+	subs, err := h.subscriptionRepo.GetSubscriptionEmailsByUserIDAndStatus(ctx, userID, domain.SubscriptionStatusSubscribed)
+	if err != nil {
+		logger.Errorf("subscriptionRepo.GetSubscriptionEmailsByUserIDAndStatus %w", err)
+		return nil, common.ErrCannotListEntity(domain.Subscription{}.DomainName(), err)
+	}
+
+	subs = append(subs, emailFromTexts...)
+	subs = util.RemoveDuplicates(subs)
```
Comment (nhan1603) [inline module/friendship/app/query/list_updates_user.go:50]: use `SELECT DISTINCT` instead

**Review by nhan1603 — CHANGES_REQUESTED** [review]

File: pkg/util/string.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    result := []string{}
+
+    for _, v := range s {
+        if !encountered[v] {
+            encountered[v] = true
+            result = append(result, v)
+        }
+    }
+    return result
+}
```
Comment (nhan1603) [inline pkg/util/string.go:24]: This function might not be necessary any more

**Comment by nhan1603** [comment]
Please perform similar check for all query using `IN` clause also.

File: module/friendship/app/query/list_updates_user.go (hunk 2 above)
Comment (phantranhieunhan) [reply module/friendship/app/query/list_updates_user.go:50]: here I just want to remove many emails already contained in text by mentions

File: module/friendship/adapter/postgres/repository/subscription.go (hunk 1 above)
Comment (phantranhieunhan) [reply module/friendship/adapter/postgres/repository/subscription.go:79]: Here, I want to join user table to get email field

**Review by nhan1603 — APPROVED** [review]

**Pull request merged**